code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -712,16 +712,17 @@
 /mob/living/simple_animal/proc/sentience_act() //Called when a simple animal gains sentience via gold slime potion
 	return
 
-/mob/living/simple_animal/update_sight()
+/mob/living/simple_animal/update_sight(reset_sight = FALSE)
 	if(!client)
 		return
 	if(stat == DEAD)
 		grant_death_vision()
 		return
 
-	see_invisible = initial(see_invisible)
-	see_in_dark = initial(see_in_dark)
-	sight = initial(sight)
+	if(reset_sight)

```

Comment:
I figured this was the best way to fix this. I couldn't find any cases in the code base where the see_invisible, see_in_dark and sight variables had to be reset on every life tick (update_sight() is called on every life tick) so in case it's needed you can now just set the first parameter to TRUE. As far as I could tell this doesn't break anything.

To make sure, @Crazylemon64 do you know if there's a reason these variables were reset every life tick?